src/Riok.Mapperly/Descriptors/MappingBuilders/EnumMappingBuilder.cs

```diff
@@ -1,134 +1,153 @@
+        if (ignoreCase)
+        {
+            var targetFieldsByNameIgnoreCase = targetFieldsByName
+                .DistinctBy(x => x.Key, StringComparer.OrdinalIgnoreCase)
+                .ToDictionary(x => x.Key, x => x.Value, StringComparer.OrdinalIgnoreCase);
+            getTargetField = source =>
+                targetFieldsByExplicitValue.GetValueOrDefault(source)
+                ?? targetFieldsByName.GetValueOrDefault(source.Name)
+                ?? targetFieldsByNameIgnoreCase.GetValueOrDefault(source.Name);
+        }
+        else
+        {
+            getTargetField = source =>
+                targetFieldsByExplicitValue.GetValueOrDefault(source) ?? targetFieldsByName.GetValueOrDefault(source.Name);
+        }
+
+        var enumMemberMappings = ctx.Source
+            .GetMembers()
+            .OfType<IFieldSymbol>()
+            .Select(x => (Source: x, Target: getTargetField(x)))
+            .Where(x => x.Target != null)
+            .ToDictionary(x => x.Source.Name, x => x.Target!.Name);
+
+        if (enumMemberMappings.Count == 0)
+        {
+            ctx.ReportDiagnostic(DiagnosticDescriptors.EnumNameMappingNoOverlappingValuesFound, ctx.Source, ctx.Target);
+        }
+
+        var missingSourceMembers = sourceFieldsByName.Where(field => !enumMemberMappings.ContainsKey(field.Key));
+        foreach (var member in missingSourceMembers)
+        {
+            ctx.ReportDiagnostic(
+                DiagnosticDescriptors.SourceEnumValueNotMapped,
+                member.Key,
+                member.Value.ConstantValue!,
+                ctx.Source,
+                ctx.Target
+            );
+        }
+
+        var missingTargetMembers = targetFieldsByName.Where(field => !enumMemberMappings.ContainsValue(field.Key));
+        foreach (var member in missingTargetMembers)
+        {
+            ctx.ReportDiagnostic(
+                DiagnosticDescriptors.TargetEnumValueNotMapped,
+                member.Key,
+                member.Value.ConstantValue!,
+                ctx.Target,
+                ctx.Source
+            );
+        }
+
+        return new EnumNameMapping(ctx.Source, ctx.Target, enumMemberMappings);
+    }
+
+    private static Dictionary<IFieldSymbol, IFieldSymbol> BuildExplicitValueMapping(MappingBuilderContext ctx)
+    {
+        var values = ctx.ListConfiguration<MapEnumValueAttribute, MapEnumValue>();
+        var targetFieldsByExplicitValue = new Dictionary<IFieldSymbol, IFieldSymbol>(SymbolEqualityComparer.Default);
+        foreach (var (sourceConstant, targetConstant) in values)
+        {
+            var source = sourceConstant.Type!.GetMembers().OfType<IFieldSymbol>().First(e => sourceConstant.Value!.Equals(e.ConstantValue));
+            var target = targetConstant.Type!.GetMembers().OfType<IFieldSymbol>().First(e => targetConstant.Value!.Equals(e.ConstantValue));
+            targetFieldsByExplicitValue.Add(source, target);

```

Comment:
This throws if a source value is specified twice, doesn't it? Instead of throwing a diagnostic would help troubleshooting bad configurations a lot.